mmv1/third_party/terraform/provider/provider.go.tmpl

```diff
@@ -37,14 +37,37 @@ func Provider() *schema.Provider {
 				Type:          schema.TypeString,
 				Optional:      true,
 				ValidateFunc:  ValidateCredentials,
-				ConflictsWith: []string{"access_token"},
+				ConflictsWith: []string{"access_token", "external_credentials"},
 			},
 
 			"access_token": {
 				Type:          schema.TypeString,
 				Optional:      true,
 				ValidateFunc:  ValidateEmptyStrings,
-				ConflictsWith: []string{"credentials"},
+				ConflictsWith: []string{"credentials", "external_credentials"},
+			},
+
+			"external_credentials": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: []string{"credentials", "access_token"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"audience": {
+							Type:     schema.TypeString,
+							Required: true, // TODO: validate audience is non-empty string
```

Comment:
are these TODO's for now or a later time?

Reply:
if we're adding validation we would want to do it out the gate

Reply:
yeah this should be included in the PR.

Reply:
this was added in this commit: https://github.com/GoogleCloudPlatform/magic-modules/pull/12714/commits/7625ed85cf5bd6b9d932087163fe13ef83072d20